demo/src/main/java/ch/srgssr/androidx/mediarouter/compose/demo/MainViewModel.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) SRG SSR. All rights reserved.
+ * License information is available from the LICENSE file.
+ */
+
+package ch.srgssr.androidx.mediarouter.compose.demo
+
+import android.app.Application
+import androidx.annotation.OptIn
+import androidx.core.net.toUri
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.media3.cast.CastPlayer
+import androidx.media3.cast.SessionAvailabilityListener
+import androidx.media3.common.MediaItem
+import androidx.media3.common.MediaMetadata
+import androidx.media3.common.MimeTypes
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.ExoPlayer
+import com.google.android.gms.cast.framework.CastContext
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted.Companion.WhileSubscribed
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(UnstableApi::class)
+class MainViewModel(application: Application) : AndroidViewModel(application) {
+    private val localPlayer = ExoPlayer.Builder(application).build()
+    private val castPlayer = CastPlayer(CastContext.getSharedInstance(application))
+    private val currentPlayer =
+        MutableStateFlow(if (castPlayer.isCastSessionAvailable) castPlayer else localPlayer)
```

Comment:
At this point in the code, castPlayer.isCastSessionAvailable is always false. It is better to check if a session is active instead.

Reply:
Actually, the `remoteMediaClient` (which is what `isCastSessionAvailable()` checks) is initialized in the `CastPlayer` constructor: https://github.com/androidx/media/blob/51efcad6720d3016d4b17fe8a1e9a11ca89e8b67/libraries/cast/src/main/java/androidx/media3/cast/CastPlayer.java#L276-L279

So if a session exists, it should be detected automatically, no?

Reply:
Yes it could work if SessionAvailabilityListener has been set within the constructor. But this listener has to be set after the constructor has been called with `CastPlayer.setSessionAvailabilityListener`.

Reply:
It has to work, my bad

https://github.com/androidx/media/blob/51efcad6720d3016d4b17fe8a1e9a11ca89e8b67/libraries/cast/src/main/java/androidx/media3/cast/CastPlayer.java#L311C3-L313C4